server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceService.java

```diff
@@ -239,7 +239,7 @@ boolean updateDesiredBalanceAndReroute(DesiredBalanceInput desiredBalanceInput,
         }
 
         final var unassigned = new HashMap<ShardId, Integer>();
-        for (var ignored : routingAllocation.routingNodes().unassigned().ignored()) {
+        for (var ignored : routingNodes.unassigned().ignored()) {
```

Comment:
I think we use `routingAllocation.routingNodes()` in other places above too - may as well fix them all?